### PR TITLE
NOOS-721/jupyterlab-templates-extension

### PR DESCRIPTION
**Pull request by TheTimb**
- Add jupyterlab_templates to environment.yml
- Fix some versions to ensure compatibility with jupyterlab<4
- Add jupyter_notebook_config.py file for templates extension configuration (modification of a directory name reflected in Dockerfile)
- Add fetchNews=false flag to remove jupyter news popup at startup in overrides.json

https://noosenergy.atlassian.net/browse/NOOS-721

### CODE REVIEW

File: docker/jupyterlab/Dockerfile [hunk 1]
```diff
@@ -24,9 +24,9 @@ RUN mamba env update -q -f /tmp/environment.yml && \
 COPY lab/config/. ${CONDA_DIR}/etc/jupyter/labconfig
 COPY lab/data/. ${CONDA_DIR}/share/jupyter/lab/settings
 
-# Configure nbconvert
-# -------------------
-COPY nbconvert/. ${HOME}/.jupyter
+# Configure nbconvert & jupyterlab_templates
```
Comment (pieeri) [inline docker/jupyterlab/Dockerfile:27]: 🐼 

Comment to be reworded maybe around the line "Configure Jupyterlab extensions" as per official docs:

https://jupyterlab.readthedocs.io/en/stable/user/directories.html#jupyterlab-user-settings-directory

**Review by pieeri — CHANGES_REQUESTED** [review]
🏓

File: docker/jupyterlab/config_extensions/jupyter_notebook_config.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+c = get_config()
+
+#c.JupyterLabTemplates.allowed_extensions = ["*.ipynb"]
```
Comment (pieeri) [inline docker/jupyterlab/config_extensions/jupyter_notebook_config.py:3]: 🐼 

Annotated code to be removed if config isn't used

File: docker/jupyterlab/lab/data/overrides.json [hunk 3]
```diff
@@ -16,6 +16,10 @@
   "@jupyterlab/apputils-extension:themes": {
     "theme": "JupyterLab Light"
   },
+  "@jupyterlab/apputils-extension:notification": {
+    "doNotDisturbMode": "true",
+    "fetchNews": "false"
```
Comment (chribag) [inline docker/jupyterlab/lab/data/overrides.json:21]: nice one !

File: docker/jupyterlab/config_extensions/jupyter_notebook_config.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+c = get_config()
```
Comment (chribag) [inline docker/jupyterlab/config_extensions/jupyter_notebook_config.py:1]: This file should be similar in structure to the nbconvert one:

# Configuration file for jupyter-nbconvert.

# more info:
# https://nbconvert.readthedocs.io/en/latest/config_options.html
# https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/config.html


We could consider merging the 2 files, but I find having both like here much clearer indeed.

**Review by chribag — CHANGES_REQUESTED** [review]

File: docker/jupyterlab/config_extensions/jupyter_notebook_config.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#------------------------------------------------------------------------------
+# Jupyter-templates configuration
+#------------------------------------------------------------------------------
```
Comment (chribag) [inline docker/jupyterlab/config_extensions/jupyter_notebook_config.py:3]: 🐼 format and wording not totally consistent with nbconvert file !

**Review by chribag — DISMISSED** [review]

**Review by pieeri — CHANGES_REQUESTED** [review]
Thanks for the changes!

Could we piggybag such a PR to:

* upgrade the version of our CircleCI Orb to the latest one (v0.2.3)
https://circleci.com/developer/orbs/orb/noosenergy/noos-ci

* upgrade the terms of our open source license (2024)

**Review by pieeri — APPROVED** [review]
👍

**Pull request merged**